metals/src/main/scala/scala/meta/internal/rename/RenameProvider.scala

```diff
@@ -147,9 +147,8 @@ final class RenameProvider(
     fileChanges
       .find { file =>
         isOccurence(str => {
-          (str.desc.isType || str.desc.isTerm) && file.endsWith(
-            str.desc.name.value + ".scala"
-          )
+          (str.desc.isType || str.desc.isTerm) &&
+            Paths.get(file).filename == str.desc.name.value + ".scala"
```

Comment:
Using just `s"/${str.desc.name.value}.scala"` will also work since it's a uri, but not sure which is the best way.

Reply:
Ah, I wasn't sure. I assumed we would get the full path here, e.g. `/src/main/scala/foo/FooBar.scala`, is that not the case?

Reply:
It should be a full uri: `file://<path-to-workspace>/a/src/main/scala/foo/FooBar.scala`

You might need to use `Paths.get(new URI(file)))`, though not 100% sure

Reply:
`Paths.get` has an overload that accepts strings. Here's my quick test in the REPL:

```scala
scala> import java.nio.file.Paths
import java.nio.file.Paths

scala> Paths.get("/foo/bar/baz/Foo.scala").getFileName().toString
res1: String = Foo.scala
```

(our `.filename` is short for `.getFileName().toString`)